dotCMS/src/main/java/com/dotcms/concurrent/DotConcurrentFactory.java

```diff
@@ -736,6 +736,13 @@ public void waitForAll(long timeout, TimeUnit unit) throws ExecutionException {
             }
         }
 
+        @Override
+        public void waitForAll() throws ExecutionException {
+            while(executorService.isTerminated()) {
+                waitForAll(10, TimeUnit.MINUTES);
```

Comment:
10 minutes is not too long?

Reply:
it is 10 minutes to check again, if the task finish before it is awake